app/models/collection_preference.rb

```diff
@@ -1,6 +1,6 @@
 class CollectionPreference < ApplicationRecord
   belongs_to :collection
-  after_update :after_update
+  after_update :after_update, :update_collection_index
```

Comment:
Should be `after_commit` for the new ES callback.

Then `update_collection_index` can skip if `destroyed?`, because of the has-one relationship between collection and preference:

https://github.com/otwcode/otwarchive/blob/253c3edfc9a428798a5bbecf95ee9741399df414/app/models/collection.rb#L22

(I'm not sure if `collection_id` will be nullified in this case.)